lib/reline/line_editor.rb

```diff
@@ -692,13 +697,6 @@ def add_dialog_proc(name, p, context = nil)
 
   DIALOG_DEFAULT_HEIGHT = 20
 
-  private def padding_space_with_escape_sequences(str, width)
```

Comment:
I'm curious, why did you combine calculating padding with `take_mbchar_range`?

Reply:
Padding sometimes have background colors.
The last testcase I added is a good example.
```
assert_equal ["\e[41m \e[42mい\e[43m ", 1, 4], Reline::Unicode.take_mbchar_range("\e[41mあ\e[42mい\e[43mう\e[0m", 1, 4, padding: true)
```
![colored padding](https://github.com/ruby/reline/assets/1780201/7fc96dc0-27ae-495a-8d57-3dfc70738f12)

So we can't add non-colored space as a padding. Padding logic should be combined with `take_mbchar_range`